common/metrics/defs.go

```diff
@@ -493,6 +508,12 @@ var MetricDefs = map[ServiceIdx]map[int]metricDefinition{
 		ConcurrencyUpdateFailureCounter:           {metricName: "concurrency-update-failure", metricType: Counter},
 		CadenceErrShardOwnershipLostCounter:       {metricName: "cadence.errors.shard-ownership-lost", metricType: Counter},
 		CadenceErrEventAlreadyStartedCounter:      {metricName: "cadence.errors.event-already-started", metricType: Counter},
+		ActivityTimerTypeHeartbeatCounter:         {metricName: "activity-timer-heartbeat", metricType: Counter},
```

Comment:
since these counters are emitted within the ActivityTimeout scope anyways, the activity-timer prefix is redundant and kinda confusing. How about renaming these to clearly indicate these are counts of specific types of timeouts i.e. heartbeat-timeouts, schedule-to-start-timeouts etc...